File: hw/pastor/u2/hw19 Inheritance/Javascript/ec.edu.espe.farmsystem.model/sheep.js
const FarmAnimal = require('./farmAnimal');

class Sheep extends FarmAnimal {
    constructor(lastSheering, id, breed, bornOn, gender, isAbleToReproduce, weight, cage, location) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage, location);
        this.lastSheering = lastSheering; // Date object
    }

    shear() {
        this.lastSheering = new Date();
    }

    toString() {
        const shearingStr = this.lastSheering.toISOString().split('T')[0];
        return super.toString() + `
 > Last Sheering : ${shearingStr}
========================================`;
    }
}

module.exports = Sheep;

File: hw/santorum/u2/hw19Inheritance/Python/model/sheep.py
from model.farmAnimal import FarmAnimal
from datetime import date

class Sheep(FarmAnimal):
    def __init__(self, lastSheering, id, breed, bornOn, gender, isAbleToReproduce, weight, cage):
        super().__init__(id, breed, bornOn, gender, isAbleToReproduce, weight, cage)
        self.lastSheering = lastSheering

    def shear(self):
        self.lastSheering = date.today()

    def __str__(self):
        return (
            f"=== SHEEP ===\n"
            f"Last Sheering: {self.lastSheering}\n"
            f"{super().__str__()}\n"
            f"============================"
        )
